docs: Improve PR template for clarity: (#7)
y-scope/public-repo-template: .github/PULL_REQUEST_TEMPLATE.md

<!-- markdownlint-disable MD012 -->

> <!--

Reviewer: deleting this and moving the PR title check to the bottom of the checklist might burry the requirement for careless readers. shall we consider keeping it?

```suggestion
<!--
Set the PR title to a meaningful commit message that follows the Conventional Commits
specification (see the checklist below for details).

Example:

fix: Don't add implicit wildcards ('*') at the beginning and the end of a query (fixes #390).
-->
```

Author: I've noticed most new developers seem to skip this comment anyway, lol. And we do have the workflow to error on an invalid commit type, so that's why I thought putting it in the checklist would be fine?

<!-- Describe what this request will change/fix and provide any details necessary for reviewers. -->

# Breaking changes

<!-- Indicate any changes that break behavior/compatibility with the last release. Delete this

> section if there are no breaking changes. -->

Reviewer: ```suggestion
<!-- Indicate any changes that break behavior/compatibility with the last release. Write "None" if
there are no breaking changes. -->

None
```

# Validation performed

<!-- Describe what tests and validation you performed on the change. -->

# Checklist

<!-- Add an `x` within each `[ ]` to indicate which requirements have been satisfied. If a
requirement hasn't been satisfied, please indicate why in a comment. -->

* [ ] The PR satisfies the [contribution guidelines][yscope-contrib-guidelines].
* [ ] Necessary docs have been updated, OR no docs need to be updated.
* [ ] The [description](#description) has been filled out.

> * [ ] The [breaking changes](#breaking-changes) section has been filled out OR deleted.

Reviewer: ```suggestion
* [ ] The [breaking changes](#breaking-changes) section has been filled out.
```

* [ ] The [validation performed](#validation-performed) section has been filled out.
* The PR title:
  * [ ] follows the [Conventional Commits][conventional-commits] specification;
  * [ ] uses one of the [commit types][commit-types] from here;
  * [ ] is in imperative form (e.g., "Add new node types."); AND

>   * [ ] indicates any GitHub feature requests or issues that the PR resolves.

Reviewer: since this is a checklist, i think omitting the `AND` / `;` should not cause ambiguity

```suggestion
  * [ ] follows the [Conventional Commits][conventional-commits] specification
  * [ ] uses one of the [commit types][commit-types] from here
  * [ ] is in imperative form (e.g., "Add new node types.")
  * [ ] references any GitHub issues that the PR resolves using the `(fixes #N)` or `(resolves #N)` syntax in the title
```

<!-- markdownlint-disable MD012 -->

# Description

<!-- Describe what this request will change/fix and provide any details necessary for reviewers. -->

# Breaking changes

<!-- Indicate any changes that break behavior/compatibility with the last release. Write "None" if
there are no breaking changes. -->

None

# Validation performed

<!-- Describe what tests and validation you performed on the change. -->

# Checklist

<!-- Add an `x` within each `[ ]` to indicate which requirements have been satisfied. If a
requirement hasn't been satisfied, please indicate why in a comment. -->

* [ ] The PR satisfies the [contribution guidelines][yscope-contrib-guidelines].
* [ ] Necessary docs have been updated, OR no docs need to be updated.
* [ ] The [description](#description) has been filled out.
* [ ] The [breaking changes](#breaking-changes) section has been filled out.
* [ ] The [validation performed](#validation-performed) section has been filled out.
* The PR title:
  * [ ] follows the [Conventional Commits][conventional-commits] specification.
  * [ ] uses one of the [commit types][commit-types] from here.
  * [ ] is in imperative form (e.g., "Add new node types.").

>   * [ ] references the GitHub issues that the PR resolves (if any) using the `(fixes #N)` or
>     `(resolves #N)` syntax in the title.

Reviewer: i've been using this template for some recent submissions and found that coding agents are generally confused by this rule (which might mean human can be equally confused?), and they made up random issue numbers in the title. How about rephrasing this as:

```suggestion
  * [ ] if the PR resolves any GitHub issues, references them in the title using the `(fixes #N)`
    or `(resolves #N)` syntax.
```

i understand this language may not flow naturally with other items, but placing the "if" condition at the beginning somewhat reduces cognitive load in reading the item in my opinion.
